### PR TITLE
Fix haddock formatting for smash-core

### PR DESCRIPTION
**Pull request by L7R7**
I tried to apply the suggested fixes from #3 for the smash-core. Linking seems to work for data types with using `'`, so I didn't change these cases yet.

### CODE REVIEW

**Comment by emilypi** [comment]
@L7R7 it looks like you have to link something like `(,)` as `@(',')@` instead of `@'(,)'@`. Who knows why, but that's how it is for operators i guess?

**Comment by L7R7** [comment]
Oh, I see. I'll fix that. tbh it's the first time really writing haddock, no idea why it's working the way it does :sweat_smile:

**Comment by emilypi** [comment]
Haddock is a complete mystery to me sometimes 😂

**Comment by emilypi** [comment]
@phadej can you confirm that the formatting for these docks is correct? I can't seem to reproduce it locally with `cabal haddock` - i'm assuming this is something that requires some extra settings to link fully?

**Comment by L7R7** [comment]
Yeah, same for me. Apart from linking to infix combinators I think I got all cases

**Comment by phadej** [comment]
>  can you confirm that the formatting for these haddocks is correct?

Things in `base` like `Maybe` and `Either` should be linked to even with local `v2-haddock`. About `(,)` and `(->)` which are special, I have no idea.

**Comment by emilypi** [comment]
It is! Yay! 🎉

File: smash-core/src/Data/Wedge.hs [hunk 1]
```diff
@@ -80,35 +80,35 @@ Categorically, the 'Wedge' datatype represents the coproduct (like, 'Either')
...
 @
 'Wedge':
-                a
-                |
-Nowhere +-------+
-                |
-                b
+                  a
+                  |
+'Nowhere' +-------+
```
Comment (phadej) [inline smash-core/src/Data/Wedge.hs:93]: Note, this breaks formatting.

**Comment by emilypi** [comment]
I might just have to do some artistic rendering to handle the visuals instead of the ascii art.

**Comment by emilypi** [comment]
@L7R7 just to keep this moving along, I think your changes are great, and thank you. I'm going to do a separate, dedicated pass for the whole library in a separate PR just before release, once the other milestones are hit.

**Pull request merged**